pyproject.toml

```diff
@@ -0,0 +1,117 @@
+[build-system]
+
+requires = [
+    "setuptools>=45.0",
+    "setuptools_scm[toml]>=6.2",
+    "wheel>=0.38.1",
+]
+
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+packages = ["pyscal"]
+
+[tool.setuptools_scm]
+write_to = "pyscal/version.py"
+
+[project]
+name = "pyscal"
+description = "Generate relative permeability include files for Eclipse reservoir simulator"
+readme = "README.md"
+requires-python = ">=3.8"
+license = { file = "LICENSE" }
+authors = [
+    { name = "Håvard Berland", email = "havb@equinor.com" },
+]
+keywords = ["relative permeability", "capillary pressure", "reservoir simulation"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research/Developers",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Utilities",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Natural Language :: English",
+    "License :: OSI Approved :: GNU Lesser General Public License v3 or later (LGPLv3+)",
+]
+dynamic = ["version"]
+
+dependencies = [
+    "matplotlib",
+    "numpy<2",
+    "openpyxl",
+    "pandas",
+    "scipy",
+    "urllib3<2",
+    "xlrd",
+]
+
+[project.optional-dependencies]
+tests = [
+"autoapi",
+"black",
+"flake8",
+"hypothesis",
+"isort",
+"mypy",
+"pre-commit",
+"pytest",
+"pytest-cov",
+"pytest-mock",
+"rstcheck",
+"sphinx<7",
+"sphinx-argparse",
+"sphinx-autodoc-typehints",
+"sphinx_rtd_theme",
```

Comment:
These lack an indent, and

```text
"autoapi",
"sphinx<7",
"sphinx-argparse",
"sphinx-autodoc-typehints",
"sphinx_rtd_theme",
```
should only be needed in the `docs` optionals.

I think you can remove `pre-commit` now too, and you might consider adding `pytest-xdist` 😄

Reply:
Note that if you move the docs dependencies out of the `tests` section, you'll need to update ci to install as `pip install .[tests,docs]`, not just `pip install .[tests]`

https://github.com/equinor/pyscal/blob/c8c385b0250f6ba340b18fe96bc0d4a44cf8fea4/.github/workflows/pyscal.yml#L53

Reply:
thanks. I saw your changes in `subscript` Now it's a good time to do the changes with `pyscal` and it cuts the test time by more than 50% :+1: